src/traces/box/cross_trace_calc.js

```diff
@@ -90,22 +95,30 @@ function setPositionOffset(traceType, gd, boxList, posAxis, pad) {
     var groupgap = fullLayout[traceType + 'groupgap'];
     var padfactor = (1 - gap) * (1 - groupgap) * dPos / fullLayout[numKey];
 
-    // autoscale the x axis - including space for points if they're off the side
-    // TODO: this will overdo it if the outermost boxes don't have
-    // their points as far out as the other boxes
-    var extremes = Axes.findExtremes(posAxis, boxdv.vals, {
-        vpadminus: dPos + pad[0] * padfactor,
-        vpadplus: dPos + pad[1] * padfactor
-    });
-
+    // Find maximum trace width
+    // we baseline this at dPos
     for(i = 0; i < boxList.length; i++) {
         calcTrace = calcdata[boxList[i]];
-        // set the width of all boxes
-        calcTrace[0].t.dPos = dPos;
-        // link extremes to all boxes
+        // set the width of this box
+        // override dPos with trace.width if present
+        var thisDPos = calcTrace[0].t.dPos = (calcTrace[0].trace.width / 2) || dPos;
+        var positions = calcTrace.map(getPosition);
+        // autoscale the x axis - including space for points if they're off the side
+        // TODO: this will overdo it if the outermost boxes don't have
+        // their points as far out as the other boxes
+
+        var trace = calcTrace[0].trace;
+        var vpadminus_pos_side = (trace.pointpos <= 0) ? (-trace.pointpos) * padfactor * fullLayout[numKey] : 0;
+        var vpadplus_neg_side = (trace.pointpos >= 0) ? trace.pointpos * padfactor * fullLayout[numKey] : 0;
+        var side = calcTrace[0].trace.side;
+        var vpadminus = (side === 'positive') ? vpadminus_pos_side + thisDPos * 0.05 : (thisDPos + pad[0] * padfactor);
+        var vpadplus = (side === 'negative') ? vpadplus_neg_side + thisDPos * 0.05 : (thisDPos + pad[1] * padfactor);
+        var extremes = Axes.findExtremes(posAxis, positions, {
+            vpadminus: vpadminus,
+            vpadplus: vpadplus
```

Comment:
let's use the [`padded: true`](https://github.com/plotly/plotly.js/blob/ad937d20a83fb3d183916e715c1a3aac6cef5921/src/plots/cartesian/autorange.js#L294-L295) option instead of those adding `* 0.05` in `vpadminus` and `vpadplus` above.